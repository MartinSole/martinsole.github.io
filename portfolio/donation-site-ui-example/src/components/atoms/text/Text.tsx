interface Props {
    as?: string
    children: string
}

const Text = ({as = "p", children}: Props) => {
    const CustomTag = `${as}` as keyof JSX.IntrinsicElements;
  return (
    <CustomTag>{children}</CustomTag>
  )
}

export default Text