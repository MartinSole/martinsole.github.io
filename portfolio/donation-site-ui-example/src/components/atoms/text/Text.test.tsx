import '@testing-library/jest-dom'
import { render } from "@testing-library/react"
import Text from "./Text"

test("Renders the Text component as a span", () => {
   const {container} = render(<Text children="Text" as="span" />)
    expect(container).toMatchSnapshot()
    const span = container.querySelector('span')
    expect(span).toBeTruthy
})
