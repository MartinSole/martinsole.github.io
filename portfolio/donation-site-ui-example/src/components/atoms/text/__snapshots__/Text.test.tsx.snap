// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the Text component as a span 1`] = `
<div>
  <span>
    Text
  </span>
</div>
`;
